### PR TITLE
Com 1285

### PR DESCRIPTION
**Pull request by sushantpatil1214**

### CODE REVIEW

File: ui/app/clinical/consultation/controllers/drugOrderHistoryController.js [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
+                   drugOrderHistoryHelper, visitHistory, $translate, $rootScope, $http) {
             var DrugOrderViewModel = Bahmni.Clinical.DrugOrderViewModel;
             var DateUtil = Bahmni.Common.Util.DateUtil;
             var currentVisit = visitHistory.activeVisit;
             var activeDrugOrdersList = [];
             var prescribedDrugOrders = [];
             $scope.dispensePrivilege = Bahmni.Clinical.Constants.dispensePrivilege;
             $scope.scheduledDate = DateUtil.getDateWithoutTime(DateUtil.addDays(DateUtil.now(), 1));
-
+            console.log("tttttttttttttttttttttttttttttttttttttttttttttttttttt");
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/controllers/drugOrderHistoryController.js]: Remove unwanted Logs

File: ui/app/clinical/consultation/controllers/drugOrderHistoryController.js [hunk 2]
```diff
@@ -176,6 +183,86 @@ angular.module('bahmni.clinical')
                     $scope.consultation.drugOrdersWithUpdatedOrderAttributes[drugOrder.uuid] = drugOrder;
                 }
             };
+            $scope.discontinue = function (drugOrder) {
+                console.log("vvvvvvvvvvvvvvvvvvvvvv");
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/controllers/drugOrderHistoryController.js]: Remove Logs

File: ui/app/clinical/consultation/controllers/drugOrderHistoryController.js [hunk 3]
```diff
@@ -176,6 +183,86 @@ angular.module('bahmni.clinical')
...
+                    $scope.updateFormConditions(drugOrder);
+                }
+            };
+            $scope.undoUpdateDispenseDateView = function (drugOrder) {
+                drugOrder.isMarkedForDispenseDate = false;
+            };
+
+            $scope.isDispenseDatePresent = function (drugOrder) {
+                console.log("1234567890");
+                // drugOrder.isDispenseDatePresent = true
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/controllers/drugOrderHistoryController.js]: Remove Commented code

File: ui/app/clinical/consultation/controllers/drugOrderHistoryController.js [hunk 4]
```diff
@@ -176,6 +183,86 @@ angular.module('bahmni.clinical')
...
+            //         .catch(function(error) {
+            //             // Handle error
+            //             console.error('Error making GET request:', error);
+            // });
+            //     drugOrder.isMarkedForDispenseDate = true;
+
+            };
+            
+            $scope.showUpdateDispenseDateView = function (drugOrder) {
+                $http.get('/openmrs/ws/rest/v1/cameroonbahmni/drugDispensedate/'+drugOrder.uuid)
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/controllers/drugOrderHistoryController.js]: Add EndPoint as constant in the Endpoint Constant file

File: ui/app/clinical/consultation/controllers/drugOrderHistoryController.js [hunk 5]
```diff
@@ -176,6 +183,86 @@ angular.module('bahmni.clinical')
...
+                    .then(function(response) {
+                        // Handle success
+                        console.log("data:", response.data);
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/controllers/drugOrderHistoryController.js]: Remove logs which prints data

File: ui/app/clinical/consultation/controllers/drugOrderHistoryController.js [hunk 6]
```diff
@@ -176,6 +183,86 @@ angular.module('bahmni.clinical')
...
+                    .catch(function(error) {
+                        // Handle error
+                        console.error('Error making GET request:', error);
+            });
+                drugOrder.isMarkedForDispenseDate = true;
+
+            };
+            $scope.updateDispenseDate = function (drugOrder) {
+                
+                console.log("88888888888888888888888888888888888888888888");
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/controllers/drugOrderHistoryController.js]: remove Logs 
Also remove extra blank line

File: ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html [hunk 7]
```diff
@@ -66,14 +81,14 @@
...
-                                        </span>
+                                         </span>
                                     </div>
                                 </div>
                                 <div ng-if="drugOrder.isMarkedForDiscontinue">
                                     <div class="form fl drug-stop-container">
                                         <article class="form-field">
                                             <div class="field-attribute stop-date-label">
-                                                <label class="control-label">{{ ::'DRUG_DETAILS_STOP_DATE' | translate }}</label>
+                                                <label class="control-label">{{ ::'DRUG_DETAILS_DATE' | translate }}</label>
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html]: This change was made for testing, Just revert as it is to `DRUG_DETAILS_STOP_DATE`

File: ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html [hunk 8]
```diff
@@ -92,6 +107,34 @@
...
+                                    <div class="form fl drug-stop-container">
+                                        <article class="form-field">
+                                            <div class="field-attribute stop-date-label">
+                                                <label class="control-label">{{ ::'Dispense Date' | translate }}</label>
+                                            </div>
+                                            <div class="field-value">
+                                                <input id="date-Start" type="date" min="{{getMinDateForDiscontinue(drugOrder)}}"
+                                                       ng-model="drugOrder.dDispenseDate"
+                                                       required="required">
+                                            </div>
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html:117]: Remove commented code

File: ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html [hunk 9]
```diff
@@ -92,6 +107,34 @@
...
+                                                <textarea ng-disabled="!drugOrder.orderReasonNotesEnabled" rows="1" id="stopped-reason-non-coded" class="form-field drug-order-notes" placeholder="{{ 'MEDICATION_RECENT_TAB_STOP_NOTES' | translate}}"
+                                                          ng-model="drugOrder.orderReasonText"></textarea>
+                                            </div> -->
+                                            <!-- <div class="field-value">
+                                            <span class="time-stamp">
+                                                <span class="time">{{drugOrder.dateActivated | bahmniDate}}</span>
+                                            </span>
+                                        </div> -->
+                                            <button class="stop-drug-btn"   ng-click="updateDispenseDate(drugOrder)">
+                                                {{ ::'Save' | translate }}
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html]: Add Save in constant or use name from constant which already present there

File: ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html [hunk 10]
```diff
@@ -92,6 +107,34 @@
...
+                                            <!-- <div class="field-value">
+                                                <textarea ng-disabled="!drugOrder.orderReasonNotesEnabled" rows="1" id="stopped-reason-non-coded" class="form-field drug-order-notes" placeholder="{{ 'MEDICATION_RECENT_TAB_STOP_NOTES' | translate}}"
+                                                          ng-model="drugOrder.orderReasonText"></textarea>
+                                            </div> -->
+                                            <!-- <div class="field-value">
+                                            <span class="time-stamp">
+                                                <span class="time">{{drugOrder.dateActivated | bahmniDate}}</span>
+                                            </span>
+                                        </div> -->
+                                            <button class="stop-drug-btn"   ng-click="updateDispenseDate(drugOrder)">
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html]: Also Add validation for save button- Dispense date is required field

File: ui/app/i18n/clinical/locale_en.json [hunk 11]
```diff
@@ -217,6 +217,9 @@
     "MEDICATION_STARTED_ON": "Started on",
     "MEDICATION_STOPPED_ON": "Stopped on",
     "MEDICATION_DUE_TO": "due to",
+    "MEDICATION_LABEL_DISPENSE_DATE": "Dispense Date",
```
Comment (sushantpatil1214) [inline ui/app/i18n/clinical/locale_en.json:220]: Also, add translation in French file

File: ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html [hunk 12]
```diff
@@ -92,6 +107,34 @@
                                         </article>
                                     </div>
                                 </div>
+
+
+                                <div ng-if="drugOrder.isMarkedForDispenseDate">
+                                    <div class="form fl drug-stop-container">
+                                        <article class="form-field">
+                                            <div class="field-attribute stop-date-label">
+                                                <label class="control-label">{{ ::'Dispense Date' | translate }}</label>
```
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html]: Use Constant for `Dispense Date`

File: ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html (hunk 9 above)
Comment (sushantpatil1214) [inline ui/app/clinical/consultation/views/treatmentSections/drugOrderHistory.html]: Add Popup after save - User will understand data is saved

**Review by sushantpatil1214 — COMMENTED** [review]
Please find out the comments bellow.

**Comment by sushantpatil1214** [comment]
Run `grunt`, resolve all Grunt errors, and test with `dist`

**Review by mahaomolise92 — APPROVED** [review]
lgtm

**Pull request merged**